src/hotspot/share/opto/loopnode.cpp

```diff
@@ -49,7 +49,10 @@
 // Determine if a node is a counted loop induction variable.
 // NOTE: The method is declared in "node.hpp".
 bool Node::is_cloop_ind_var() const {
-  return (is_Phi() && !as_Phi()->is_copy() &&
+#ifdef ASSERT
+  if (is_Phi()) assert(!as_Phi()->is_copy(), "this phi cannot be a copy");
```

Comment:
The if condition should be added to the assert (the `#ifdef ASSERT` can then be removed):
`assert(!is_Phi() || !as_Phi()->is_copy() ...)`